docs/tutorials/mongoid-configuration.txt

```diff
@@ -0,0 +1,304 @@
+=============
+Configuration
+=============
+
+.. _mongoid-configuration:
+
+Mongoid is customarily configured through a ``mongoid.yml`` file that specifies
+options and clients. The simplest configuration is as follows, which configures
+Mongoid to talk to a MongoDB server at "localhost:27017" and use the database
+named "mongoid".
+
+.. code-block:: yaml
+
+  development:
+    clients:
+      default:
+        database: mongoid
+        hosts:
+          - localhost:27017
+
+The top level key in the configuration file, ``development`` in the above
+example, refers to the environment name which the application is executing in,
+i.e. ``development``, ``test`` or ``production``. The third level key,
+``default`` in the above example, refers to the Mongo client name.
+Most applications will use a single client named ``default``.
+
+Rails Applications
+******************
+
+If you are using Rails, you can have Mongoid generate a default configuration
+file for you by running the following command:
+
+.. code-block:: bash
+
+  $ rails g mongoid:config
+
+The configuration file will be placed in ``config/mongoid.yml``.
+
+Environment Determination
+*************************
+
+When Mongoid loads its configuration, it chooses the environment to use
+in the following order:
+
+- ``Rails.env`` if using Rails.
+- ``Sinatra::Base.environment`` if using Sinatra.
+- ``RACK_ENV`` environment variable.
+- ``MONGOID_ENV`` environment variable.
+
+If you are not building a Rack based application and want to override the
+environment programatically, you can pass a second parameter to ``load!``
+as follows:
+
+.. code-block:: ruby
+
+  Mongoid.load!("path/to/your/mongoid.yml", :production)
+
+Depending on your version of Rails, you may also need to configure the ORM
+to be Mongoid in application.rb like so:
+
+.. code-block:: ruby
+
+  config.generators do |g|
+    g.orm :mongoid
+  end
+
+Anatomy of a Mongoid Config
+***************************
+
+Let's have a look at a full-blown ``mongoid.yml`` and explain the full power
+of what Mongoid can do. The following configuration has explanation of the
+various options in the comments above each key:
+
+.. code-block:: yaml
+
+  development:
+    # Configure available database clients. (required)
+    clients:
+      # Define the default client. (required)
+      default:
+        # A uri may be defined for a client:
+        # uri: 'mongodb://user:password@myhost1.mydomain.com:27017/my_db'
```

Comment:
If I recall correctly, a configuration can't have both a URI field and host/options fields. We should document this.